src/finch/tensor.py

```diff
@@ -154,12 +157,11 @@ def __mod__(self, other):
     def __pow__(self, other):
         return self._elemwise_op("^", other)
 
-    def __matmul__(self, other):
-        # TODO: Implement and use mul instead of tensordot
-        # https://github.com/FinchTensor/finch-tensor-python/pull/22#issuecomment-2007884763
-        if self.ndim != 2 or other.ndim != 2:
-            raise ValueError(
-                f"Both tensors must be 2-dimensional, but are: {self.ndim=} and {other.ndim=}."
+    @compiled()
```

Comment:
We don't need `@compiled` decorator here.

Reply:
Done in https://github.com/finch-tensor/finch-tensor-python/pull/83/commits/87f384c06d507d9c9024d880b347977fa7a14a5f.